libs/feature-workspaces/src/lib/projects/projects.component.ts

```diff
@@ -37,31 +61,64 @@ export class ProjectsComponent implements OnInit {
     shareReplay()
   );
 
+  recentActions$: Observable<ProjectActionMap>;
+
   constructor(
     private readonly route: ActivatedRoute,
     private readonly workspaceGQL: WorkspaceGQL,
     private readonly settings: Settings,
     private readonly workspaceDocsGQL: WorkspaceDocsGQL,
-    private readonly commandRunner: CommandRunner
+    private readonly saveRecentActionGQL: SaveRecentActionGQL,
+    private readonly commandRunner: CommandRunner,
+    private readonly workspaceSchematicsGQL: WorkspaceSchematicsGQL
   ) {}
 
   ngOnInit() {
-    this.projects$ = this.route.params.pipe(
+    const workspace$ = this.route.params.pipe(
       map(m => m.path),
       switchMap(path => {
-        return this.workspaceGQL.watch(
-          {
-            path
-          },
-          {
-            pollInterval: PROJECTS_POLLING
-          }
-        ).valueChanges;
+        return combineLatest(
+          this.workspaceGQL.watch(
+            {
+              path
+            },
+            {
+              pollInterval: PROJECTS_POLLING
+            }
+          ).valueChanges,
+          this.workspaceSchematicsGQL
+            .watch({ path }, { pollInterval: PROJECTS_POLLING })
+            .valueChanges.pipe(
+              catchError(() =>
+                of({
+                  data: {
+                    workspace: {
+                      schematicCollections: [] as WorkspaceSchematics.SchematicCollections[]
+                    }
+                  }
+                })
+              )
+            )
+        );
       }),
-      map((r: any) => {
-        const w = r.data.workspace;
-        const projects = w.projects.map((p: any) => {
-          return { ...p, actions: this.createActions(p) };
+      filter(([r1, r2]) => !!r1 && !!r2),
```

Comment:
Boolean(r1 && r2) is a bit clearer in my opinion

Reply:
Ok